base/files/file_starboard.cc

```diff
@@ -455,7 +455,17 @@ void File::DoInitialize(const FilePath& path, uint32_t flags) {
   file_.reset(descriptor);
 
   if (!file_.is_valid()) {
+#if defined(__ANDROID_API__)
```

Comment:
This looks a bit hacky, but if it works around the current issues it's fine to go in